basyx.submodelrepository/basyx.submodelrepository-http/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/http/SubmodelRepositoryApiHTTPController.java

```diff
@@ -65,176 +61,101 @@ public SubmodelRepositoryApiHTTPController(SubmodelRepository repository) {
-	public ResponseEntity<List<Submodel>> getAllSubmodels(
-			@Parameter(in = ParameterIn.QUERY, description = "The value of the semantic id reference (BASE64-URL-encoded)", schema = @Schema()) @Valid @RequestParam(value = "semanticId", required = false) String semanticId,
-			@Parameter(in = ParameterIn.QUERY, description = "The Submodel’s idShort", schema = @Schema()) @Valid @RequestParam(value = "idShort", required = false) String idShort) {
-		return new ResponseEntity<List<Submodel>>(new ArrayList<>(repository.getAllSubmodels()), HttpStatus.OK);
+	public ResponseEntity<Submodel> postSubmodel(@Parameter(in = ParameterIn.DEFAULT, description = "Submodel object", required = true, schema = @Schema()) @Valid @RequestBody Submodel body) {
+		repository.createSubmodel(body);
+		return new ResponseEntity<Submodel>(body, HttpStatus.CREATED);
 	}
 
 	@Override
-	public ResponseEntity<?> getSubmodelSubmodelRepo(
-			@Parameter(in = ParameterIn.PATH, description = "The Submodel’s unique id (BASE64-URL-encoded)", required = true, schema = @Schema()) @PathVariable("submodelIdentifier") Base64UrlEncodedIdentifier submodelIdentifier,
-			@Parameter(in = ParameterIn.QUERY, description = "Determines the structural depth of the respective resource content", schema = @Schema(allowableValues = { "deep",
-					"core" }, defaultValue = "deep")) @Valid @RequestParam(value = "level", required = false, defaultValue = "deep") String level,
-			@Parameter(in = ParameterIn.QUERY, description = "Determines the request or response kind of the resource", schema = @Schema(allowableValues = { "normal", "trimmed", "value", "reference",
-					"path" }, defaultValue = "normal")) @Valid @RequestParam(value = "content", required = false, defaultValue = "normal") String content,
-			@Parameter(in = ParameterIn.QUERY, description = "Determines to which extent the resource is being serialized", schema = @Schema(allowableValues = { "withBlobValue",
-					"withoutBlobValue" })) @Valid @RequestParam(value = "extent", required = false) String extent) {
-		if (isNormalContentRequest(content)) {
-			return new ResponseEntity<Submodel>(repository.getSubmodel(submodelIdentifier.getIdentifier()), HttpStatus.OK);
-		} else if (isValueContentRequest(content)) {
-			return new ResponseEntity<SubmodelValueOnly>(new SubmodelValueOnly(repository.getSubmodelElements(submodelIdentifier.getIdentifier())), HttpStatus.OK);
-		} else if (isMetadataContentRequest(content)) {
-			Submodel submodel = repository.getSubmodel(submodelIdentifier.getIdentifier());
-			submodel.setSubmodelElements(null);
-			return new ResponseEntity<Submodel>(submodel, HttpStatus.OK);
-		}
-		return new ResponseEntity<Object>(HttpStatus.NOT_IMPLEMENTED);
+	public ResponseEntity<Void> deleteSubmodelById(
+			@Parameter(in = ParameterIn.PATH, description = "The Submodel’s unique id (BASE64-URL-encoded)", required = true, schema = @Schema()) @PathVariable("submodelIdentifier") Base64UrlEncodedIdentifier submodelIdentifier) {
```

Comment:
``BASE64-URL-encoded`` was changed at some places to ``UTF8-BASE64-URL-encoded`` &rarr; Why not here as well?